### PR TITLE
Filter: Distinguish between .so and .socket

### PR DESCRIPTION
**Pull request by athvu**
Make sure that a package with a file name containing .so is actually
a shared object file and not socket.

Current implementation of shared object filtering did not take into
account .socket files, so packages that should've been filtered out were
not.

Signed-off-by: Angelina Vu <vuangelina72@gmail.com>

### CODE REVIEW

File: binaryaudit/abicheck.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ def filter_rpm(filename, filter_list, rpm, drop_count):
         has_so = False
         for member in rpm.getmembers():
             member_name = str(member)
-            if ".so" in member_name:
+            if ".so" in member_name and ".socket" not in member_name:
```
Comment (weltling) [inline binaryaudit/abicheck.py:227]: Probably `endswith` is what should be used.

Thanks

Comment (weltling) [reply binaryaudit/abicheck.py:227]: One more comment - a stronger check would be also adding `abicheck.is_elf()` to the condition, as that would ensure a file ending with `.so` is actually an ELF artifact. Not sure if it's doable at this stage though, as seems here it just lists the contents of an RPM.

Thanks

Comment (athvu) [reply binaryaudit/abicheck.py:227]: I did try it with endswith, but it did not work since sometimes member_name would end in .so followed by some numbers. For example, /usr/lib/libyajl.so.2.1.0

Comment (weltling) [reply binaryaudit/abicheck.py:227]: Ah, for that case, i think using a regex might be more reliable then. Please consider this:

```python3
import re
fname = "/usr/lib/libyajl.so.2.1.0"
so_reg = re.compile(r".*\.so(\.\d*)*$")
if so_reg.match(fname):
    print("got DSO\n")
```

There might be some edge cases, so perhaps putting it into a function and having a test would be best to serve this piece. Fe it should match `/usr/lib/libyajl.so.2.1.0` and `/usr/lib/libyajl.so`, but shouldn't match `libyajl.socket` or `/var/run/docker.sock` or alike.

Thanks

Comment (weltling) [reply binaryaudit/abicheck.py:227]: Ups, the pattern would be probably `.*\.so(\.\d+)*$` as the number is not optional.

Perhaps introducing something like `util.is_dso_filename(filename)` and adding a test would be the way?

Thanks

**Review by weltling — APPROVED** [review]
